Kitodo/src/main/java/org/kitodo/services/image/MissingImagesFilterPredicate.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * (c) Kitodo. Key to digital objects e. V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo project.
+ *
+ * It is licensed under GNU General Public License version 3 or later.
+ *
+ * For the full copyright and license information, please read the
+ * GPL3-License.txt file that was distributed with this source code.
+ */
+
+package org.kitodo.services.image;
+
+import java.util.function.Predicate;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.kitodo.data.database.beans.Folder;
+import org.kitodo.model.Subfolder;
+
+public class MissingImagesFilterPredicate implements Predicate<Subfolder> {
+    private static final Logger logger = LogManager.getLogger(MissingImagesFilterPredicate.class);
+
+    /**
+     * This message is written to the logfile if the file was found.
+     */
+    private static final String IMAGE_FOUND = "Image {0} was found in folder {1}.";
+
+    /**
+     * This message is written to the logfile if the file was not found.
+     */
+    private static final String IMAGE_NOT_FOUND = "Image {0} not found in folder {1}: Marked for generation.";
+
+    /**
+     * The canonical part of the filename. Usually this is the base name without
+     * an extension, but in some special cases, it can only be part of the base
+     * name. This is configured in the folder and separated in the content
+     * folder.
+     */
+    private String canonical;
+
+    /**
+     * Creates a filter predicate that checks if the file exists in the folder.
+     * The name of the file results from the settings of the folder passed into
+     * the {@link #test(Folder)} function, and the canonical name part and the
+     * variables.
+     * 
+     * @param canonical
+     *            the canonical part of the file name
+     */
+    public MissingImagesFilterPredicate(String canonical) {
+        this.canonical = canonical;
+    }
+
+    /**
+     * Check if there is a corresponding file in the folder. The name of the
+     * file results from the settings of the folder, and the canonical name part
+     * and the variables passed in the constructor.
+     * 
+     * @return true, if the picture needs to be generated
+     */
+    @Override
+    public boolean test(Subfolder destination) {
```

Comment:
Please name the method more narrative. What is tested? why is it tested?

Reply:
I can not do that. The class implements the Predicate interface and the interface defines what the method should be called. It is tested whether there is a matching file in the given folder to the canonical part of the file name transferred during the creation of the object. If not, true is returned to enable the folder to be generated for this derivative.

Reply:
Is it necessary to make this class as an implementation of Predicate?

Reply:
Yes, otherwise I would not have done it.